src/sentry/issues/auto_source_code_config/constants.py

```diff
@@ -1,17 +1,28 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
 SUPPORTED_INTEGRATIONS = ["github"]
-# XXX: We may want to change these constants into a configuration object
-# Any new languages should also require updating the stacktraceLink.tsx and repo_trees.py SUPPORTED_EXTENSIONS
-SUPPORTED_LANGUAGES = [
-    "csharp",
-    "go",
-    "javascript",
-    "node",
-    "php",
-    "python",
-    "ruby",
-]
-# Some languages will also process system frames
-PROCESS_ALL_FRAMES: list[str] = ["java"]
-# Extract filename from module and abs_path
-# e.g. com.foo.bar.Baz$handle$1, Baz.kt -> com/foo/bar/Baz.kt
-EXTRACT_FILENAME_FROM_MODULE_AND_ABS_PATH = ["java"]
+
+
+# Any new languages should also require updating the stacktraceLink.tsx
+# The extensions do not need to be exhaustive but only include the ones that show up in stacktraces
+PLATFORMS_CONFIG: dict[str, Mapping[str, Any]] = {
+    "csharp": {"extensions": ["cs"]},
+    "go": {"extensions": ["go"]},
+    "java": {
+        # e.g. com.foo.bar.Baz$handle$1, Baz.kt -> com/foo/bar/Baz.kt
+        "extract_filename_from_module": True,
+        "create_in_app_stack_trace_rules": True,
```

Comment:
Renamed this from `process_all_frames` since the reason we process all frames is to mark them as in_app.